app/code/Magento/NewsletterGraphQl/Model/Resolver/SubscribeEmailToNewsletter.php

```diff
@@ -0,0 +1,163 @@
+     * @var CustomerRepositoryInterface
+     */
+    private $customerRepository;
+
+    /**
+     * @var EmailValidator
+     */
+    private $emailValidator;
+
+    /**
+     * @var ScopeConfigInterface
+     */
+    private $scopeConfig;
+
+    /**
+     * @var SubscriberResourceModel
+     */
+    private $subscriberResource;
+
+    /**
+     * @var SubscriptionManagerInterface
+     */
+    private $subscriptionManager;
+
+    /**
+     * SubscribeEmailToNewsletter constructor.
+     *
+     * @param CustomerRepositoryInterface $customerRepository
+     * @param EmailValidator $emailValidator
+     * @param ScopeConfigInterface $scopeConfig
+     * @param SubscriberResourceModel $subscriberResource
+     * @param SubscriptionManagerInterface $subscriptionManager
+     */
+    public function __construct(
+        CustomerRepositoryInterface $customerRepository,
+        EmailValidator $emailValidator,
+        ScopeConfigInterface $scopeConfig,
+        SubscriberResourceModel $subscriberResource,
+        SubscriptionManagerInterface $subscriptionManager
+    ) {
+        $this->customerRepository = $customerRepository;
+        $this->emailValidator = $emailValidator;
+        $this->scopeConfig = $scopeConfig;
+        $this->subscriberResource = $subscriberResource;
+        $this->subscriptionManager = $subscriptionManager;
+    }
+
+    /**
+     * Adds an email into a newsletter subscription
+     *
+     * @param Field $field
+     * @param ContextInterface $context
+     * @param ResolveInfo $info
+     * @param array|null $value
+     * @param array|null $args
+     * @return array
+     * @throws GraphQlInputException
+     */
+    public function resolve(
+        Field $field,
+        $context,
+        ResolveInfo $info,
+        array $value = null,
+        array $args = null
+    ) {
+        $email = trim($args['email']);
+
+        if (empty($email)) {
+            throw new GraphQlInputException(__('Specify the "email" value.'));
+        }
+
+        try {
+            // Validate the format of the email address
+            if (!$this->emailValidator->isValid($email)) {
+                throw new GraphQlInputException(__('Please enter a valid email address.'));
+            }
+
+            // Validate that if the current user is a guest, that they can subscribe to a newsletter.
+            if (false === $context->getExtensionAttributes()->getIsCustomer()
+                && !$this->scopeConfig->getValue(
+                    Subscriber::XML_PATH_ALLOW_GUEST_SUBSCRIBE_FLAG,
+                    ScopeInterface::SCOPE_STORE
+                )
+            ) {
+                throw new GraphQlInputException(
+                    __('Sorry, but the administrator denied subscription for guests. Please register.')
+                );
+            }
+
+            // Validate if a provided email is not equal to the customer's one
+            $currentUserId = $context->getUserId();
+            if ($currentUserId > 0) {
```

Comment:
> The current logic in the resolver denies using any email address for the logged-in customer except the account primary account email address. Please, correct me if I'm wrong.

Correct. 

If a logged in customer (_for example, with `customer@server.com` email_) provides a different email address (_for example, `some_customer@server.com`_) then it will get `Registered customer can not subscribe an email of another user.` error.

> According to the current core implementation, an existing email should not be allowed if the existing email does not belong to the currently logged in user (refer to the [original](https://github.com/magento/magento2/blob/f060ee7d46a5a88b97c846c1c37fd3d3c447c32d/app/code/Magento/Newsletter/Controller/Subscriber/NewAction.php#L91) method for details)

I've provided the same logic in the resolver.

Thank you!

Reply:
**fixed**